### PR TITLE
 #1195 feat: pagination in config

### PR DESCRIPTION
**Pull request by didoda**
This provides pagination configurable

### CODE REVIEW

File: plugins/BEdita/API/src/Controller/AppController.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function initialize()
 
         $this->response = $this->response->withHeader('X-BEdita-Version', Configure::read('BEdita.version'));
 
-        $this->loadComponent('BEdita/API.Paginator');
+        $this->loadComponent('BEdita/API.Paginator', Configure::read('Pagination'));
```
Comment (stefanorosanelli) [inline plugins/BEdita/API/src/Controller/AppController.php]: If no `Pagination` key is found in config an empty array must be passed

(see travis errors)

Also a sample setting in `app.default.php` is missing

File: plugins/BEdita/API/src/Controller/Component/PaginatorComponent.php [hunk 2]
```diff
@@ -34,12 +34,15 @@ class PaginatorComponent extends CakePaginatorComponent
         'whitelist' => ['page', 'page_size', 'sort'],
     ];
 
+    public $absoluteLimit = 500;
```
Comment (stefanorosanelli) [inline plugins/BEdita/API/src/Controller/Component/PaginatorComponent.php]: I would use `protected` to avoid possible easy changes....

**Review by stefanorosanelli — CHANGES_REQUESTED** [review]

**Review by stefanorosanelli — APPROVED** [review]

File: plugins/BEdita/API/src/Controller/Component/PaginatorComponent.php [hunk 3]
```diff
@@ -34,12 +34,15 @@ class PaginatorComponent extends CakePaginatorComponent
         'whitelist' => ['page', 'page_size', 'sort'],
     ];
 
+    protected $absoluteLimit = 500;
```
Comment (fquffio) [inline plugins/BEdita/API/src/Controller/Component/PaginatorComponent.php]: Docblock is missing for this class member. 🙃 

However, I'd probably use `const MAX_LIMIT = 500;` instead.

File: plugins/BEdita/API/src/Controller/AppController.php [hunk 4]
```diff
@@ -55,7 +55,11 @@ public function initialize()
 
         $this->response = $this->response->withHeader('X-BEdita-Version', Configure::read('BEdita.version'));
 
-        $this->loadComponent('BEdita/API.Paginator');
+        $pagination = Configure::read('Pagination');
+        if (empty($pagination)) {
+            $pagination = [];
+        }
+        $this->loadComponent('BEdita/API.Paginator', $pagination);
```
Comment (fquffio) [inline plugins/BEdita/API/src/Controller/AppController.php]: (This is a _veeeeery minor_ issue…)

In order to reduce complexity (which makes Scrutinizer complain at times) we could avoid the `if` statement by just writing:
```php
$this->loadComponent('BEdita/API.Paginator', (array)Configure::read('Pagination'));
```

If that configuration key is missing, `Configure::read()` returns `null`, and casting NULL to an array results in an empty array (`(array)null === []`), which is perfect. 😎

**Review by fquffio — APPROVED** [review]

File: plugins/BEdita/API/src/Controller/AppController.php (hunk 4 above)
Comment (didoda) [reply plugins/BEdita/API/src/Controller/AppController.php]: :+1:

**Pull request merged**